packages/cli/src/agentInstaller/pythonAgentInstaller.js

```diff
@@ -35,30 +34,33 @@ class PoetryInstaller extends BuildToolInstaller {
    * @returns {string}
    */
   get assumptions() {
-    return `Your project contains a ${POETRY_LOCK_FILE}. Therefore, it looks like a poetry project,
-so we will install the AppMap Python package "appmap".`;
+    return null;
```

Comment:
It seems more consistent to me if this step is required and states why the flow is choosing this path (because a poetry lock file exists). 

Then the install command is run and the post install message is shown. Then a confirmation command is also run to verify the install - which may be a formality, since the success of the install command is technically enough. But it’s helpful to the user, I think, to always have a confirmation step?

Reply:
If we make this step required, the user sees the confirmation message:

```
Press enter to continue, 'a' abort, or 'm' to run it yourself manually:
```

which doesn't make sense here.

I think if we want to completely rework this flow, it should probably happen in a separate PR?

Reply:
Why doesn't it make sense?

Reply:
Assumptions was designed to always be not-null, because it's explaining what path the installer has chosen - information that the user will always want to have.

Reply:
But what does "run it yourself manually" mean without an action? Isn't that going to be confusing?

I hear what you're saying about `assumptions` providing information the user needs, though. AS we move forward, I think we'll want to make the API a little more flexible so it can accommodate the requirements of each of the languages.

Reply:
It would probably mean:

Developer: "I let you try and do this once, but you mangled my file, so I understand what we are trying to do here, and I am going to do it myself, then let you continue with the rest of your flow".